src/FTPClientWrapperSSH.cpp

```diff
@@ -553,47 +552,19 @@ int FTPClientWrapperSSH::authenticate_key(ssh_session session) {
 	}
 
 	char * keyfile = SU::TCharToCP(m_keyFile, CP_ACP);
-	ssh_private_key privkey = NULL;
-	if (m_passphrase[0] == 0)	//in case the passphrase is empty, use NULL instead
-		privkey = privatekey_from_file(session, keyfile, 0, NULL);
-	else
-		privkey = privatekey_from_file(session, keyfile, 0, m_passphrase);
+	ssh_key privkey = NULL;
+	//in case the passphrase is empty, use NULL instead
+	rc = ssh_pki_import_privkey_file(keyfile, m_passphrase[0] ? m_passphrase : NULL, NULL, NULL, &privkey);

```

Comment:
I committed a change which removes the call to PathFileExists and improves error logging. This should resolve #135.

I also added code to show the host key type to satisfy #54. There is certainly more that can be done but this change should satisfy the user.

The issue with #141 seems to be a lack of support for diffie-hellman-group-exchange-sha256 in libssh. That's just a guess though. More info is needed from the user.